src/components/TodoItems.tsx

```diff
@@ -1,125 +1,100 @@
 import classNames from 'classnames';
 import React, { useEffect, useRef, useState } from 'react';
-import { deleteTodo, updateTodo } from '../api/todos';
+import { updateTodo } from '../api/todos';
 import { Todo } from '../types/Todo';
-import { ErrorList } from '../types/ErrorList';
+import { ErrorMessages } from '../types/ErrorMessages';
 import { useTodosContext } from '../context/TodoContext';
 
 interface Props {
   todo: Todo;
 }
 
 export const TodoItem: React.FC<Props> = ({ todo }) => {
-  const {
-    processingIds,
-    focusInput,
-    todos,
-    setTodos,
-    handleError,
-    handleDeleteTodo,
-  } = useTodosContext();
+  const { processingIds, setTodos, handleError, handleDeleteTodo } =
+    useTodosContext();
 
   const [queryEditing, setQueryEditing] = useState(todo.title);
   const [isLoading, setIsLoading] = useState(false);
   const [isEditing, setIsEditing] = useState(false);
+  const loaderIsActive =
+    processingIds.includes(todo.id) || todo.id === 0 || isLoading;
 
   const inputTodoRef = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    focusInput();
     if (isEditing && inputTodoRef.current) {
       inputTodoRef.current?.focus();
     }
   }, [isEditing]);
 
-  const handleChecker = async (ItemID: number) => {
-    const todosChecker = todos.map(prevItem => {
-      if (prevItem.id === ItemID) {
-        return {
-          ...prevItem,
-          completed: !prevItem.completed,
-        };
-      }
-
-      return prevItem;
-    });
-
+  const handleChecker = (itemID: number) => {
     setIsLoading(true);
 
     const newTodo = {
       ...todo,
       completed: !todo.completed,
     };
 
-    await updateTodo(ItemID, newTodo)
-      .then(() => {
-        setTodos(todosChecker);
+    updateTodo(itemID, newTodo)
+      .then(todoValue => {
+        setTodos(prevTodos =>
+          prevTodos.map(currentTodo =>
+            currentTodo.id === todo.id ? todoValue : currentTodo,
+          ),
+        );
       })
       .catch(() => {
-        handleError(ErrorList.UpdateTodo);
+        handleError(ErrorMessages.UpdateTodo);
       })
       .finally(() => {
-        focusInput();
+        inputTodoRef.current?.focus();
         setIsLoading(false);
       });
   };
 
   const saveInputValue = () => {
```

Comment:
Can we somehow rewrite this logic to avoid using `if, else if, else`? 

I believe that if we just create 3 conditionals it will be more readable.